### PR TITLE
Test that CDI Interceptors are invoked on JAX-RS resources and providers

### PR DESCRIPTION
**Pull request by andymc12**
New test for CDI interceptors - ensuring they are invoked properly on JAX-RS resource methods and provider methods.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
@jkoehler22 yes, the test suite automatically runs with EE7 features (JAX-RS 2.0 + CDI 1.2), and then repeats the tests with EE8 features (2.1 + 2.0).  Thanks for the review.

File: dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/publish/servers/com.ibm.ws.jaxrs20.cdi12.fat.interceptor/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/apps
```
[the remark this answers is not in the document]
Comment (andymc12) [reply dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/publish/servers/com.ibm.ws.jaxrs20.cdi12.fat.interceptor/.gitignore]: it is consistent with the other servers in the publish directory (put there when I copied an existing server directory).  if we don't this one, then we should probably delete all of them, right?

**Comment by andymc12** [comment]
#build

File: dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/publish/servers/com.ibm.ws.jaxrs20.cdi12.fat.interceptor/.gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/publish/servers/com.ibm.ws.jaxrs20.cdi12.fat.interceptor/.gitignore]: ah, I may be off base here. I was originally thinking this was in the root level .gitignore file. this is probably fine to leave in, but I'm still wondering if it's an actual necessity. If not, we could put cleaning up our .gitignore files on our backlog :)

**Comment by andymc12** [comment]
All failures are unrelated to these changes.  Since this is only adding new tests (and they all passed), it should be safe and self-contained.

**Pull request merged**